mailbox/maildir/src/main/java/org/apache/james/mailbox/maildir/MaildirStore.java

```diff
@@ -237,13 +241,15 @@ public String getFolderName(MailboxPath mailboxPath) {
         return getFolderName(mailboxPath.getNamespace(), mailboxPath.getUser(), mailboxPath.getName());
     }
 
-    /**
-     * @see org.apache.james.mailbox.store.mail.UidProvider#nextUid(org.apache.james.mailbox.MailboxSession, org.apache.james.mailbox.store.mail.model.Mailbox)
-     */
     @Override
-    public long nextUid(MailboxSession session, Mailbox mailbox) throws MailboxException {
+    public MessageUid nextUid(MailboxSession session, Mailbox mailbox) throws MailboxException {
         try {
-            return createMaildirFolder(mailbox).getLastUid(session) +1;
+            return createMaildirFolder(mailbox).getLastUid(session).transform(new Function<MessageUid, MessageUid>() {
```

Comment:
If all these functions would have been defined in MessageUid class, this would avoid these repetitions, no?